docs/build/circuit-construction.ipynb

```diff
@@ -0,0 +1,486 @@
+      "text/plain": [
+       "<Figure size 538.33x367.889 with 1 Axes>"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "qc_a.decompose().draw(\"mpl\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "649fc3fd-caf1-45f1-ad8e-3b5d26ca859b",
+   "metadata": {},
+   "source": [
+    "## Parameterized circuits\n",
+    "\n",
+    "Many near-term quantum algorithms repeatedly execute a circuit, then tweak the circuit based on the result. Using this process, these algoirthms find circuits that produce specific outputs. The following cell shows a very crude example of this: a loop that increases a gate angle until the circuit outputs `1` over 99% of the time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "fb8535d8-6911-48f7-ba42-ed2b9c8d3cad",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Angle: 2.949999999999998\n",
+      " P(1): 0.990851101499227\n"
+     ]
+    }
+   ],
+   "source": [
+    "from qiskit.quantum_info import Statevector  # to simulate the circuit\n",
+    "from qiskit import transpile  # to optimize the circuit\n",
+    "\n",
+    "value = .5\n",
+    "while True:\n",
+    "    # Create and optimize circuit\n",
+    "    qc = QuantumCircuit(1)\n",
+    "    qc.rx(value, 0)\n",
+    "    qc = transpile(qc)\n",
+    "\n",
+    "    # Simulate circuit\n",
+    "    p1 = Statevector(qc).probabilities()[1]\n",
+    "    if p1 > 0.99:\n",
+    "        break\n",
+    "\n",
+    "    # Tweak values\n",
+    "    value += 0.05\n",
+    "\n",
+    "print(f\"Angle: {value}\\n P(1): {p1}\")"
```

Comment:
This code example is very crude. Is it ok? Or is there something more realistic we could do that isn't any more complicated?

Reply:
maybe use the example from the advanced circuits tutorial? Generally I think there are some nice code snippets from there that we can reuse

Reply:
Also in general I think we should include an example for unbound parameters as well as one showing assigning parameters

Reply:
Thanks! Yeah the tutorial just compiles a list of circuits which I think is better idea. I've also drawn the unbound circuit too.